### DISCUSSION TITLE
多个bot在同一个群时got装饰器的matcher获取的bot不符合预期

### DISCUSSION THREAD

**Discussion by jerry-harm** [nonebot/discussions#528] [问答] [answered]
我试图用一个nonebot服务器控制两个bot在同一个群里实现不同的功能，但是当触发其中一个bot时如果用了got方法，matcher有概率使用另一个bot进行消息发送，不知道这个是否是可以控制的

**Comment by yanyongyu** [comment, marked as answer]
由于两个bot在同一个群，一个事件会收到两次。你可以通过permission updater来自定义会话流程中的下一次响应条件，检查event中的self_id。
